### PR TITLE
Use 961$i field to get vendor Id

### LINKED ISSUE #31: Map 961$i to vendor reference number
Map `961$i` to vendor reference number with type of Internal vendor number.

### PR DESCRIPTION
**Pull request by fereira**
Use 961$i field to get vendor Id for vendorDetail

resolves #31

### CODE REVIEW

File: src/main/java/org/olf/folio/order/OrderImport.java [hunk 1]
```diff
@@ -232,15 +233,19 @@ public  JSONArray  upload(String fileName) throws IOException, InterruptedExcept
 				locations.put(location);
 				 
 				
-				//VENDOR REFERENCE NUMBER IF INCLUDED IN THE MARC RECORD:
+				// as of IRIS release vendorDetail is slightly more complex
 				if (vendorItemId != null) {
```
Comment (cappadona) [inline src/main/java/org/olf/folio/order/OrderImport.java:237]: @fereira For some reason this test is true for any file I submit, even if there are no records with a `961$i`. Are you seeing the same?

File: src/main/java/org/olf/folio/order/OrderImport.java [hunk 2]
```diff
@@ -232,15 +233,19 @@ public  JSONArray  upload(String fileName) throws IOException, InterruptedExcept
...
-					vendorDetail.put("instructions", "");
-					vendorDetail.put("refNumber", vendorItemId);
-					vendorDetail.put("refNumberType", "Internal vendor number");
-					vendorDetail.put("vendorAccount", "");
-					orderLine.put("vendorDetail", vendorDetail);
-				}
+                    JSONArray referenceNumbers = new JSONArray();
+                    JSONObject vendorDetail = new JSONObject();
+                    vendorDetail.put("instructions", "");
+                    vendorDetail.put("vendorAccount", "");
```
Comment (cappadona) [inline src/main/java/org/olf/folio/order/OrderImport.java:241]: I know these were both already here, but no harm in dropping them if we're not setting values, correct?

```suggestion
```

**Comment by fereira** [comment]
MarcUtils.getVendorItemId now returns null if a 961 field is missing and OrderImport and MarcToJson test if empty as well as if null

File: src/main/java/org/olf/folio/order/util/MarcUtils.java [hunk 3]
```diff
@@ -103,6 +103,8 @@ public String getVendorItemId(DataField nineSixtyOne ) {
         String vendorItemId= new String();
         if (nineSixtyOne != null) {
             vendorItemId = nineSixtyOne.getSubfieldsAsString(VENDORID);
+        } else {
+            return null;
```
Comment (cappadona) [inline src/main/java/org/olf/folio/order/util/MarcUtils.java:107]: Just for my own benefit, shouldn't `vendorItemId` be `null` when initialized on line 103?

**Review by cappadona — APPROVED** [review]
Thanks @fereira. This is now working as expected 👏🏼

File: src/main/java/org/olf/folio/order/util/MarcUtils.java (hunk 3 above)
Comment (fereira) [reply src/main/java/org/olf/folio/order/util/MarcUtils.java:107]: I am considering revisiting all of the MarcUtils methods and how they're tested.  In this case, the method is returning null if the 961 field or the $i subfield is missing.  It could also return an empty string if the 961 field and $i subfield exists, but contains an empty string.  Although some methods return a default value (e.g. "olin" for getLocation
